libcxx/test/std/utilities/expected/expected.void/observers/value.pass.cpp

```diff
@@ -65,17 +65,6 @@ void testException() {
     }
   }
 
-  // MoveOnly
```

Comment:
Out of interest, according to the LWG issue, this test should never work. Do you know why there was not an issue before?

Reply:
Yeah due to LWG3940 it cannot pass the compiler. But I guess before this the `expected::value` implementation can actually accept move-only types.

Reply:
> Out of interest, according to the LWG issue, this test should never work. Do you know why there was not an issue before?

This is because of Clang's non-conforming acception of move-only exception types, which is reported as #53224.

Reply:
Why would this extension be non-conforming? A warning is missing, but other than that I don't see how it wouldn't be conforming.

Reply:
It seems that `throw NonCopyable(args)` also needs result in substitution failure, while Clang currently disagrees ([example](https://godbolt.org/z/x1j16E54K)). See also [this article](https://quuxplusone.github.io/blog/2021/03/18/sfinae-on-throw-x/).